bundles/org.openhab.binding.modbus.kermi/src/main/java/org/openhab/binding/modbus/kermi/internal/handler/KermiXcenterThingHandler.java

```diff
@@ -0,0 +1,666 @@
+            WorkHoursDTO dto = (WorkHoursDTO) dtoOpt.get();
+            updateState(workHoursFanChannel, dto.workHoursFan);
+            updateState(workHoursStorageLoadingPumpChannel, dto.workHoursStorageLoadingPump);
+            updateState(workHoursCompressorChannel, dto.workHoursCompressor);
+        } else {
+            logger.debug("Unable to get {} from provider {}", Data.DataType.WORK_HOURS, workHoursParser.toString());
+        }
+    }
+
+    private void handleWorkHoursFailure(AsyncModbusFailure<ModbusReadRequestBlueprint> result) {
+        if (workHoursRead != ReadStatus.READ_FAILED) {
+            // update status only if bit switches
+            workHoursRead = ReadStatus.READ_FAILED;
+            logger.debug("Cause of failure: {}", result.getCause().getMessage());
+            updateStatus();
+        }
+    }
+
+    @Override
+    public void dispose() {
+        ModbusCommunicationInterface localComms = comms;
+        if (localComms != null) {
+            for (PollTask p : pollTasks) {
+                PollTask localPoller = p;
+                if (localPoller != null) {
+                    localComms.unregisterRegularPoll(localPoller);
+                }
+            }
+        }
+        // Comms will be close()'d by endpoint thing handler
+        comms = null;
+    }
+
+    private void updateStatus() {
+        logger.debug("Status update: State {} Data {} WorkHours {} PV {} ", stateRead, powerRead, workHoursRead,
+                pvRead);
+        if (stateRead != ReadStatus.NOT_RECEIVED) { // && dataRead != ReadStatus.NOT_RECEIVED
+            if (stateRead == ReadStatus.READ_SUCCESS) {
+                updateStatus(ThingStatus.ONLINE);
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        KermiBindingConstants.STATE_READ_ERROR);
+            }
+            if (stateRead == alarmRead) {
+                // both reads are ok or else both failed
+                if (stateRead == ReadStatus.READ_SUCCESS) {
+                    updateStatus(ThingStatus.ONLINE);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            KermiBindingConstants.STATE_AND_ALARM_READ_ERRORS);
+                }
+            } else {
+                // either info or data read failed - update status with details
+                if (stateRead == ReadStatus.READ_FAILED) {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            KermiBindingConstants.STATE_READ_ERROR);
+                } else {
+                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                            KermiBindingConstants.DATA_READ_ERROR);
+                }
+            }
+        } // else - one status isn't received yet - wait until both Modbus polls returns either success or error
```

Comment:
Do you need to do anything with this case? Use it or remove it

Reply:
should work as it is now.